system/cpu.go

```diff
@@ -286,39 +298,44 @@ func GetFLInfo() (string, string, bool) {
 	cpuStateDiffer := false
 	cpuStateMap := make(map[string]string)
 
+	if !supportsForceLatencySettings("") {
+		return "all:none", "all:none", cpuStateDiffer
+	}
+
 	// read /sys/devices/system/cpu
 	dirCont, err := ioutil.ReadDir(cpuDir)
-	if runtime.GOARCH != "ppc64le" && err == nil {
-		// latency settings are only relevant for Intel-based systems
-		for _, entry := range dirCont {
-			// cpu0 ... cpuXY
-			if isCPU.MatchString(entry.Name()) {
-				// read /sys/devices/system/cpu/cpu*/cpuidle
-				cpudirCont, err := ioutil.ReadDir(path.Join(cpuDir, entry.Name(), "cpuidle"))
-				if err != nil {
-					// idle settings not supported for entry.Name()
-					continue
-				}
-				supported = true
-				for _, centry := range cpudirCont {
-					// state0 ... stateXY
-					if isState.MatchString(centry.Name()) {
-						// read /sys/devices/system/cpu/cpu*/cpuidle/state*/disable
-						state, _ := GetSysString(path.Join(cpuDirSys, entry.Name(), "cpuidle", centry.Name(), "disable"))
-						// save latency states for 'revert'
-						// savedStates = "cpu1:state0:0 cpu1:state1:0"
-						savedStates = savedStates + " " + entry.Name() + ":" + centry.Name() + ":" + state
-						cpuStateMap[entry.Name()] = cpuStateMap[entry.Name()] + " " + state
-						// read /sys/devices/system/cpu/cpu*/cpuidle/state*/latency
-						lattmp, _ := GetSysInt(path.Join(cpuDirSys, entry.Name(), "cpuidle", centry.Name(), "latency"))
-						if lattmp > maxlat {
-							maxlat = lattmp
-						}
-						if state == "1" {
-							stateDisabled = true
-						} else {
-							lat = lattmp
-						}
+	if err != nil {
+		WarningLog("latency settings not supported by the system")
+		return "all:none", "all:none", cpuStateDiffer
+	}
+	for _, entry := range dirCont {
+		// cpu0 ... cpuXY
+		if isCPU.MatchString(entry.Name()) {
```

Comment:
Why don't create a variable for `entry.Name()`? `entry.Name()` will be called multiple time and this can increase the time computing of the block because entry.Name() have is own time to compute. We are probably talking about insignificant nanoseconds, but you never know :)

Reply:
yes, good idea. See new commit